toucan_connectors/http_api/authentication_configs.py

```diff
@@ -0,0 +1,251 @@
+import logging
+from abc import ABC, abstractmethod
+from datetime import datetime, timedelta
+from typing import Any, Callable, Literal
+from urllib import parse as url_parse
+
+from pydantic import BaseModel, ConfigDict, Field, SecretStr, ValidationError
+from requests import Session
+
+from toucan_connectors.common import UI_HIDDEN
+from toucan_connectors.json_wrapper import JsonWrapper
+from toucan_connectors.oauth2_connector.oauth2connector import oauth_client
+
+_LOGGER = logging.getLogger(__name__)
+
+
+try:
+    # Module can be missing if toucan-connectors is installed in light mode
+    # Those missing modules must be also included to HttpApiConnector module dependencies
+    from authlib.common.security import generate_token
+except ImportError:
+    _LOGGER.warning("Missing dependencies for HttpApi Connector authentication")
+
+
+class Oauth2Error(Exception):
+    """Base class for oauth2 related errors"""
+
+
+class MissingRefreshTokenError(Oauth2Error):
+    """Raised when refresh token is missing from saved token"""
+
+
+class MissingOauthWorkflowError(Oauth2Error):
+    """Raised when the oauth2 workflow is missing from saved session."""
+
+
+_SUPPORTED_EXPIRATION_FORMATS = ["%Y-%m-%d %H:%M:%S", "%Y-%m-%dT%H:%M:%S", "%d/%m/%Y %H:%M:%S"]
+
+
+def _extract_expires_at_from_token_response(token_response: dict[str, Any], default_lifetime: int) -> datetime:
+    """Read token expiration info from oauth2 dance response and return a computed datetime.
+
+    It can raise an exception if the expiration date has un unexpected format.
+    """
+    if "expires_at" in token_response:
+        if isinstance(token_response["expires_at"], str):
+            expires_at = None
+            for date_format in _SUPPORTED_EXPIRATION_FORMATS:
+                try:
+                    expires_at = datetime.strptime(token_response["expires_at"], date_format)
+                    break
+                except ValueError:
+                    continue
+            if expires_at is not None:
+                return expires_at
+            raise ValueError(f"Can't parse the oauth2 token expiration: {token_response['expires_at']}")
+        return datetime.fromtimestamp(int(token_response["expires_at"]))
+    else:
+        return datetime.now() + timedelta(0, int(token_response.get("expires_in", default_lifetime)))
```

Comment:
note: `dateutil`'s `timedelta` is smarter. Also, you you please use kwargs for the timedelta ?